src/main/java/io/vertx/core/http/impl/ConnectionManager.java

```diff
@@ -124,16 +124,16 @@ void getConnection(ContextInternal ctx, String peerHost, boolean ssl, int port,
             if (metrics != null) {
               metrics.closeEndpoint(host, port, metric);
             }
-            endpointMap.remove(key); // Should remove with current endpoint value to avoid races
+            endpointMap.remove(key);
           },
-          connectionMap::put,
-          connectionMap::remove,
+          conn -> connectionMap.put(conn.channel(), conn),
+          conn -> connectionMap.remove(conn.channel(), conn),
           false);
         return new Endpoint(pool, metric);
       });
       Object metric;
       if (metrics != null) {
-        metric = metrics.enqueueRequest(endpoint.metric); // Metrics might be enqueued several times if getConnection returns false
+        metric = metrics.enqueueRequest(endpoint.metric);
```

Comment:
The comment was added in one the first commits. Was it before the logic was refactored and now the risk has been mitigated? I'm asking because I'm not sure why this comment was added and now removed without the surrounding code being modified.

Reply:
it was before